### DISCUSSION TITLE
jMolecules Layers Arch Tests

### DISCUSSION THREAD

**Discussion by Tobias-08** [spring-projects/spring-modulith#852] [Q&A] [answered]
Hi,
how do I activate the jMolecules Layers Arch Tests? If I reference jMolecules the DDD tests are executed automatically (and fail). Are the DDD tests hard coded or can I switch to layers by config?

Thanks!

**Comment by odrotbohm** [comment, marked as answer]
On current Spring Modulith versions, you'd create a dedicated test case and issue `JMoleculesArchitectureRules.ensureLayering().check(classes)`. As of 1.3 RC1, we will automatically execute the tests if you have the corresponding jMolecules architecture library on the classpath. See GH-862 for details.

**Reply by Tobias-08** [reply]
Thanks! GH-862 is the direction I was hoping for.
